### ISSUE TITLE
AppStore publishing - AppIcon missing

### ISSUE THREAD

**Issue by shaiscytale** [dotnet/maui#22421]
Hello !

When trying to distribute my app with transporter, I'm stuck for many days on asset validation with the following error :

"Asset validation failed (90704) Missing app icon. Include a large app icon as a 1024 by 1024 pixel PNG in the asset catalog of apps built for iOS, iPadOS, or watchOS. Without this icon, apps can't be submitted for review...."

I tried many workarounds, switching to svg, then png, then svg again, create an asset group with many file sizes, go back to simpliest implementation...

I'm using Visual studio 17.9.2 (rollback from more recent version due to incompatibility with XCode), a macbook air with sonoma 14.4 and Xcode 15.3.

The build works fine, and the .csproj looks fine too. I don't know if it's really an issue, or an error in my project configuration.

the .csproj : 

```xml
<Project Sdk="Microsoft.NET.Sdk.Razor">

	<PropertyGroup>
		<TargetFrameworks>net7.0-android;net7.0-ios;</TargetFrameworks>
		<OutputType>Exe</OutputType>
		<RootNamespace>Ces.Mobile</RootNamespace>
		<UseMaui>true</UseMaui>
		<SingleProject>true</SingleProject>
		<ImplicitUsings>enable</ImplicitUsings>
		<EnableDefaultCssItems>false</EnableDefaultCssItems>
		<nullable>enable</nullable>

		<!-- Display name -->
		<ApplicationTitle>ApplicationTitle</ApplicationTitle>

		<!-- App Identifier -->
		<ApplicationId>ApplicationId</ApplicationId>
		<ApplicationIdGuid>ApplicationIdGuid</ApplicationIdGuid>

		<!-- Versions -->
		<ApplicationDisplayVersion>1.0.33</ApplicationDisplayVersion>
		<ApplicationVersion>72</ApplicationVersion>

		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">14.2</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">29.0</SupportedOSPlatformVersion>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-android|AnyCPU'">
		<AndroidUseAapt2>True</AndroidUseAapt2>
		<AndroidCreatePackagePerAbi>False</AndroidCreatePackagePerAbi>
		<AndroidPackageFormat>apk</AndroidPackageFormat>
		<EmbedAssembliesIntoApk>False</EmbedAssembliesIntoApk>
		<EnableLLVM>False</EnableLLVM>
		<IpaPackageName>IpaPackageName</IpaPackageName>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-android|AnyCPU'">
		<AndroidPackageFormat>aab</AndroidPackageFormat>
		<AndroidUseAapt2>True</AndroidUseAapt2>
		<AndroidCreatePackagePerAbi>False</AndroidCreatePackagePerAbi>
		<EmbedAssembliesIntoApk>True</EmbedAssembliesIntoApk>
		<EnableLLVM>False</EnableLLVM>
		<PublishTrimmed>True</PublishTrimmed>
		<RunAOTCompilation>True</RunAOTCompilation>
		<IpaPackageName>IpaPackageName</IpaPackageName>
	</PropertyGroup>

	<PropertyGroup Condition="'$(TargetFramework)'=='net7.0-ios'">
		<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
		<ProvisioningType>manual</ProvisioningType>
		<CodesignKey>CodesignKey</CodesignKey>
		<ApplicationId>ApplicationId</ApplicationId>
		<AdditionalArgs>-v -v -v -v --linkskip=System.ComponentModel.TypeConverter</AdditionalArgs>
		<MtouchDebug>True</MtouchDebug>
		<MtouchLink>None</MtouchLink>
		<BuildIpa>True</BuildIpa>
		<IpaPackageName>IpaPackageName</IpaPackageName>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-ios|AnyCPU'">
		<CodeSignEntitlements>Platforms/iOS/Entitlements-dev.plist</CodeSignEntitlements>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-ios|AnyCPU'">
		<CodeSignEntitlements>Platforms/iOS/Entitlements.plist</CodeSignEntitlements>
		<CodesignKey>CodesignKey</CodesignKey>
		<CodesignProvision>CodesignProvision</CodesignProvision>
	</PropertyGroup>

	
	<ItemGroup>
        <MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.svg" Color="#512BD4" />

		<MauiSplashScreen Include="Resources\Splash\splash.svg" Color="#FF6161" BaseSize="80,80" Resize="true" />

		<MauiImage Include="Resources\Images\*" />

		<MauiFont Include="Resources\Fonts\*" />

        <MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
      	
	</ItemGroup>

	<ItemGroup>
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon1024.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon120.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon152.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon167.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon180.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon20.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon29.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon40.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon58.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon60.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon76.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon80.png" />
		<None Remove="Assets.xcassets\AppIcon.appiconset\Icon87.png" />
		<None Remove="Platforms\Android\Resources\xml\network_security_config.xml" />
	</ItemGroup>

	<ItemGroup Condition="'$(Configuration)' == 'Debug'">
		<EmbeddedResource Include="appsettings.Development.json" />
		<Content Remove="appsettings.Development.json" />
		<EmbeddedResource Include="appsettings.iOSDevelopment.json" />
		<Content Remove="appsettings.iOSDevelopment.json" />
	</ItemGroup>
	<ItemGroup Condition="'$(Configuration)' == 'Release'">
		<EmbeddedResource Include="appsettings.json" />
		<Content Remove="appsettings.json" />
	</ItemGroup>

	<ItemGroup Condition="'$(TargetFramework)'=='net7.0-android'">
		<Content Remove="wwwroot\google-services.json" />
		<GoogleServicesJson Update="wwwroot\google-services.json">
			<CopyToOutputDirectory>Always</CopyToOutputDirectory>
		</GoogleServicesJson>
	</ItemGroup>

	<ItemGroup>
		<PackageReference Include="Blazored.Modal" Version="7.1.0" />
		<PackageReference Include="Blazored.Toast" Version="3.2.2" />
		<PackageReference Include="BlazorStrap" Version="5.0.106" />
		<PackageReference Include="Microsoft.AspNetCore.Components.Authorization" Version="6.0.12" />
		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="7.0.2" />
		<PackageReference Include="Microsoft.Identity.Client" Version="4.55.0" />
		<PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="2.32.0" />
		<PackageReference Include="Plugin.Maui.Audio" Version="2.1.0" />
		<PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.35.0" />
	</ItemGroup>

	<ItemGroup Condition="'$(TargetFramework)' == 'net7.0-android'">
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Common" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData.Core" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData.Core.Ktx" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Process" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel.Ktx" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModelSavedState" Version="2.7.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Collection" Version="1.4.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.4.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Fragment.Ktx" Version="1.6.2.2" />
		<PackageReference Include="Xamarin.AndroidX.Collection.Jvm" Version="1.4.0.1" />
		<PackageReference Include="Xamarin.AndroidX.Activity" Version="1.8.2.1" />
		<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.2.1" />
		<PackageReference Include="Xamarin.Firebase.Core" Version="121.1.1.6" />
		<PackageReference Include="Xamarin.Firebase.Messaging" Version="123.3.1.1" />
		<PackageReference Include="Xamarin.GooglePlayServices.Base" Version="118.2.0.5" />
		<PackageReference Include="Xamarin.Google.Dagger" Version="2.48.1.1" />
		<PackageReference Include="Plugin.Firebase.CloudMessaging" Version="2.0.4" />
	</ItemGroup>

	<ItemGroup>
		<ProjectReference Include="..\ApiClient\ApiClient.csproj" />
		<ProjectReference Include="..\Contracts\Contracts.csproj" />
		<ProjectReference Include="..\Domain.Enumerations\Domain.Enumerations.csproj" />
		<ProjectReference Include="..\Domain\Domain.csproj" />
		<ProjectReference Include="..\Mobile.States\Mobile.States.csproj" />
		<ProjectReference Include="..\States.Abstraction\States.Abstraction.csproj" />
	</ItemGroup>

	<ItemGroup>
		<EditorConfigFiles Remove="C:\Projects\CES\Ces\Mobile.Presentation\Components\Procedure\.editorconfig" />
	</ItemGroup>

	<ItemGroup>
	  <GoogleServicesJson Include="wwwroot\google-services.json">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </GoogleServicesJson>
	</ItemGroup>

	<ItemGroup>
		<None Include="C:\Projects\CES\Ces\Mobile.Presentation\Components\Procedure\.editorconfig" />
	</ItemGroup>

	<ProjectExtensions>
		<VisualStudio>
			<UserProperties appsettings_1development_1json__JsonSchema="https://geojson.org/schema/GeoJSON.json" XamarinHotReloadDebuggerTimeoutExceptionMobileHideInfoBar="True" XamarinHotReloadGenericExceptionInfoBarMobileHideInfoBar="True" />
		</VisualStudio>
	</ProjectExtensions>

	<ItemGroup>
		<BundleResource Include="Assets.xcassets\**\*" />  
	</ItemGroup>

	<ItemGroup>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon1024.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon120.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon152.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon167.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon180.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon20.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon29.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon40.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon58.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon60.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon76.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon80.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	  <Content Include="Assets.xcassets\AppIcon.appiconset\Icon87.png">
	    <CopyToOutputDirectory>Always</CopyToOutputDirectory>
	  </Content>
	</ItemGroup>

</Project>
```

and a view of my project folder : 

![image](https://github.com/dotnet/maui/assets/49271112/cdf13f5d-6cc2-4b57-931f-155e1289cc3e)

**Comment by markmccaigue** [comment]
Hi @shaiscytale it might be worth checking your Info.plist file if you haven't already?

More info below:
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/images/app-icons?view=net-maui-8.0&tabs=macios#tabpanel_1_macios

**Comment by shaiscytale** [comment]
Hi ! the info.plist is pretty basic, and I added the required "XSAppIconAssets" part

`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
		</dict>
	</dict>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>UIDeviceFamily</key>
	<array>
		<integer>1</integer>
	</array>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>arm64</string>
	</array>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
	</array>
	<key>UISupportedInterfaceOrientations~ipad</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationPortraitUpsideDown</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>CFBundleIdentifier</key>
	<string>CFBundleIdentifier</string>
	<key>CFBundleURLTypes</key>
	<array>
		<dict>
			<key>CFBundleURLName</key>
			<string>CFBundleURLName</string>
			<key>CFBundleURLSchemes</key>
			<array>
				<string>CFBundleURLSchemes</string>
			</array>
			<key>CFBundleTypeRole</key>
			<string>Editor</string>
		</dict>
	</array>
	<key>CFBundleShortVersionString</key>
	<string>1.0.33</string>
	<key>XSAppIconAssets</key>
	<string>Assets.xcassets/appicon.appiconset</string>
	<key>NSCalendarsUsageDescription</key>
	<string>This app needs access to your calendar.</string>
	<key>NSContactsUsageDescription</key>
	<string>This app needs access to your contact list.</string>
	<key>NSMicrophoneUsageDescription</key>
	<string>This app needs access to your microphone to record audio messages.</string>
	<key>NSAppleMusicUsageDescription</key>
	<string>This app needs access to your music.</string>
	<key>NSSiriUsageDescription</key>
	<string>This app needs access to Siri.</string>
	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
	<string>This app needs access to your location.</string>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string>This app needs access to your location.</string>
	<key>NSBluetoothAlwaysUsageDescription</key>
	<string>This app needs access to your bluetooth.</string>
	<key>NSCameraUsageDescription</key>
	<string>This app needs access to your camera to take pictures.</string>
	<key>CFBundleVersion</key>
	<string>72</string>
</dict>
</plist>
`

**Comment by mattleibow** [comment]
May be related #12085 

TL;DR:

Set `/p:IpaIncludeArtwork=true` on the CLI or `<IpaIncludeArtwork>true</IpaIncludeArtwork>` in the csproj.

**Comment by shaiscytale** [comment]
Hi @mattleibow  !

the <IpaIncludeArtwork>true</IpaIncludeArtwork> seems to work for ipa distribution, but since I'm no longer able to debug for an unknown reason, I'm still investigating. Thanks anyway !

**Comment by jfversluis** [comment]
Original issue has been resolved, thanks all!
